compiler/lib/src/main/scala/codegen/CppWriter/ComponentCppWriter/ComponentParameters.scala

```diff
@@ -58,36 +59,62 @@ case class ComponentParameters (
       addAccessTagAndComment(
         "PRIVATE",
         "Parameter validity flags",
-        sortedParams.map((_, param) =>
-          linesClassMember(
-            lines(
-              s"""|
-                  |//! True if ${param.getName} was successfully received
-                  |Fw::ParamValid ${paramValidityFlagName(param.getName)};
-                  |"""
+        sortedParams.flatMap { case (_, param) =>
+          if (param.isExternal) then Nil
+          else List(
+            linesClassMember(
+              lines(
+                s"""|
+                    |//! True if ${param.getName} was successfully received
+                    |Fw::ParamValid ${paramValidityFlagName(param.getName)};
+                    |"""
+              )
             )
           )
-        ),
+        },
         CppDoc.Lines.Hpp
       ),
       addAccessTagAndComment(
         "PRIVATE",
         "Parameter variables",
-        sortedParams.map((_, param) => {
-          val paramType = writeParamType(param.paramType, "Fw::ParamString")
-          val paramVarName = paramVariableName(param.getName)
-          linesClassMember(
-            List.concat(
-              addSeparatedPreComment(
-                s"Parameter ${param.getName}",
-                AnnotationCppWriter.asStringOpt(param.aNode)
-              ),
-              lines(s"$paramType $paramVarName;")
+        sortedParams.flatMap { case (_, param) =>
+          if (param.isExternal) then Nil
+          else {
+            val paramType = writeParamType(param.paramType, "Fw::ParamString")
+            val paramVarName = paramVariableName(param.getName)
+            List(
+              linesClassMember(
+                List.concat(
+                  addSeparatedPreComment(
+                    s"Parameter ${param.getName}",
+                    AnnotationCppWriter.asStringOpt(param.aNode)
+                  ),
+                  lines(s"$paramType $paramVarName;")
+                )
+              )
             )
-          )
-        }),
+          }
+        },
         CppDoc.Lines.Hpp
-      )
+      ),
+      guardedList (hasExternalParameters) (
+        List(
+          addAccessTagAndComment(
+            "PRIVATE",
+            "Parameter delegates",
+            List(
+              linesClassMember(
+                lines(
+                  s"""|
+                      |//! Delegate to serialize/deserialize an externally stored parameter
+                      |Fw::ParamExternalDelegate* paramDelegatePtr;
+                      |"""
+                )
+              )
+            )
+          )
+        )
+      ).flatten
```

Comment:
This is a very minor point, but we have switched from writing `List(...).flatten` to `List.concat(...)`. It's slightly clearer.